include/semiprof/semiprof.hpp

```diff
@@ -45,11 +45,17 @@ std::ostream& operator<<(std::ostream&, const profile&);
 
     // leave a profling region
     #define PL semiprof::profiler_leave
+    #define PP() std::cout << semiprof::profiler_summary() << "\n"
```

Comment:
I don't think macros are appropriate for clearing and printing the profiler, which would be an infrequent operation.
The `PL` and `PE` are for conevenience.

Can we use C style function calls instead, and make the calls do `nullopt` when compiled without profiler support?

Reply:
Thanks Ben for the review! Our use case is that we execute some function multiple times and would like to know the breakdown of time after the "warm-up", which is why we reset the profiler with each iteration.

I agree that the C-style function calls might be nicer, this will be changed.

Reply:
The purpose of the macros is to automate awkward and repetitive patterns used to make entering and leaving profiler regions fast. The `PP` macro does not fit that pattern: it produces very specifically formatted output in `std::cout`, which isn't something I think that the library should provide as a default.

Reply:
Agreed! Should be removed now.